packages/ssi-types/src/mapper/credential-mapper.ts

```diff
@@ -293,6 +293,82 @@ export class CredentialMapper {
     return '@context' in credential && ((credential as ICredential).type?.includes('VerifiableCredential') || false)
   }
 
+  public static isCredential(original: OriginalVerifiableCredential | OriginalVerifiablePresentation): boolean {
+    let document = original
+    try {
+      if (typeof original === 'string') {
+        if (this.isJwtEncoded(original)) {
+          const vc: IVerifiableCredential = this.toUniformCredential(original)
+          return this.isW3cCredential(vc)
+        } else if (this.isSdJwtEncoded(original)) {
+          // find out how we can differentiate between vc and vp for sd-jwts
+          return true
+        } else {
+          document = JSON.parse(original)
+        }
+      }
+      return (
+        this.isW3cCredential(document as ICredential) ||
+        this.isSdJwtDecodedCredentialPayload(document as ICredential) ||
+        this.isJwtDecodedCredential(document as OriginalVerifiableCredential) ||
+        this.isSdJwtDecodedCredential(document as OriginalVerifiableCredential)
+      )
+    } catch (e) {
+      return false
+    }
+  }
+
+  public static isPresentation(original: OriginalVerifiableCredential | OriginalVerifiablePresentation): boolean {
+    let document = original
+    try {
+      if (typeof original === 'string')
+        if (this.isJwtEncoded(original)) {
+          const vp: IVerifiablePresentation = this.toUniformPresentation(original)
+          return this.isW3cPresentation(vp)
+        } else if (this.isSdJwtEncoded(original)) {
+          return true
```

Comment:
Hmzz, this needs to be documented. Because an SD JWT VC is used as a presentation as well. Right now both the isCredential and isPresentation methods return true for the same passed in SD JWT. If the context of calling this method is to make a distinction between VCs and VPs then the isPresentation should return false. If however the context is to determine from code whether we are constructing or using a VP somewhere in the code, that it would be okay to return true here, because although SD JWTs are always VCs, they are also used as presentations.

My suggestion is to change this one to false. And then to adjust the logic in where the isPresentation and isCredential is currently called. If it needs a presentation somewhere, then you can just use true in case it is a SD-JWT

Reply:
changed this to `false` for now, but we have to have a discussion about `sd_jwt`s and how are we going to support these. right now it's partially supported and in some cases it will lead to unwanted results.